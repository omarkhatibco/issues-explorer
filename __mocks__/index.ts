export * from './fail';
export * from './response';
export * from './singleResponse';
