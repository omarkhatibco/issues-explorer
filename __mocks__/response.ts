import { SearchQueryResponse } from 'types';

interface Response {
	data: SearchQueryResponse;
}

export const response: Response = {
	data: {
		search: {
			issueCount: 542,
			pageInfo: {
				hasNextPage: true,
				hasPreviousPage: false,
				__typename: 'PageInfo'
			},
			nodes: [
				{
					number: 21695,
					title: '[DevTools Bug]: Long keys overlap',
					state: 'OPEN',
					closedAt: null,
					createdAt: '2021-06-16T20:57:00Z',
					comments: {
						totalCount: 1,
						__typename: 'IssueCommentConnection'
					},
					author: {
						login: 'MinimumViablePerson',
						__typename: 'User'
					},
					labels: {
						totalCount: 4,
						nodes: [
							{
								id: 'MDU6TGFiZWw3MTA1NzM1OTU=',
								name: 'Component: Developer Tools',
								color: 'fbca04',
								__typename: 'Label'
							},
							{
								id: 'MDU6TGFiZWw2MjAzNjg0MDc=',
								name: 'Resolution: Needs More Information',
								color: 'fffde7',
								__typename: 'Label'
							},
							{
								id: 'MDU6TGFiZWwxNTU5ODQxNjA=',
								name: 'Status: Unconfirmed',
								color: 'd4c5f9',
								__typename: 'Label'
							},
							{
								id: 'MDU6TGFiZWw0MDkyOTE1MQ==',
								name: 'Type: Bug',
								color: 'b60205',
								__typename: 'Label'
							}
						],
						__typename: 'LabelConnection'
					},
					__typename: 'Issue'
				},
				{
					number: 21693,
					title: 'Allow customization of Dom validation',
					state: 'OPEN',
					closedAt: null,
					createdAt: '2021-06-16T17:27:00Z',
					comments: {
						totalCount: 0,
						__typename: 'IssueCommentConnection'
					},
					author: {
						login: 'stevemk14ebr',
						__typename: 'User'
					},
					labels: {
						totalCount: 1,
						nodes: [
							{
								id: 'MDU6TGFiZWwxNTU5ODQxNjA=',
								name: 'Status: Unconfirmed',
								color: 'd4c5f9',
								__typename: 'Label'
							}
						],
						__typename: 'LabelConnection'
					},
					__typename: 'Issue'
				},
				{
					number: 21692,
					title: 'Bug: memory leak in useReducer',
					state: 'OPEN',
					closedAt: null,
					createdAt: '2021-06-16T16:11:17Z',
					comments: {
						totalCount: 1,
						__typename: 'IssueCommentConnection'
					},
					author: {
						login: 'onnlucky',
						__typename: 'User'
					},
					labels: {
						totalCount: 1,
						nodes: [
							{
								id: 'MDU6TGFiZWwxNTU5ODQxNjA=',
								name: 'Status: Unconfirmed',
								color: 'd4c5f9',
								__typename: 'Label'
							}
						],
						__typename: 'LabelConnection'
					},
					__typename: 'Issue'
				},
				{
					number: 21685,
					title: 'Bug: Webpack process and setImmediate polyfill interferes with time slicing',
					state: 'OPEN',
					closedAt: null,
					createdAt: '2021-06-15T17:07:37Z',
					comments: {
						totalCount: 0,
						__typename: 'IssueCommentConnection'
					},
					author: {
						login: 'gaearon',
						__typename: 'User'
					},
					labels: {
						totalCount: 3,
						nodes: [
							{
								id: 'MDU6TGFiZWwxMjA1MDg3MTI3',
								name: 'Component: Concurrent Features',
								color: 'ffccd3',
								__typename: 'Label'
							},
							{
								id: 'MDU6TGFiZWwzMDc5MDUzOTE5',
								name: 'React 18',
								color: '26B0E4',
								__typename: 'Label'
							},
							{
								id: 'MDU6TGFiZWw0MDkyOTE1MQ==',
								name: 'Type: Bug',
								color: 'b60205',
								__typename: 'Label'
							}
						],
						__typename: 'LabelConnection'
					},
					__typename: 'Issue'
				},
				{
					number: 21674,
					title: 'React 18: react-router@v5 is breaking in the Strict Mode (strict effects)',
					state: 'OPEN',
					closedAt: null,
					createdAt: '2021-06-14T03:43:07Z',
					comments: {
						totalCount: 5,
						__typename: 'IssueCommentConnection'
					},
					author: {
						login: 'Jack-Works',
						__typename: 'User'
					},
					labels: {
						totalCount: 2,
						nodes: [
							{
								id: 'MDU6TGFiZWwzMDc5MDUzOTE5',
								name: 'React 18',
								color: '26B0E4',
								__typename: 'Label'
							},
							{
								id: 'MDU6TGFiZWw3MTAzNzU3OTI=',
								name: 'Type: Discussion',
								color: 'fef2c0',
								__typename: 'Label'
							}
						],
						__typename: 'LabelConnection'
					},
					__typename: 'Issue'
				},
				{
					number: 21668,
					title: 'Question: Transition starvation in React 18?',
					state: 'OPEN',
					closedAt: null,
					createdAt: '2021-06-12T10:25:53Z',
					comments: {
						totalCount: 0,
						__typename: 'IssueCommentConnection'
					},
					author: {
						login: 'Ephem',
						__typename: 'User'
					},
					labels: {
						totalCount: 2,
						nodes: [
							{
								id: 'MDU6TGFiZWwzMDc5MDUzOTE5',
								name: 'React 18',
								color: '26B0E4',
								__typename: 'Label'
							},
							{
								id: 'MDU6TGFiZWw3MTAzNzU3OTI=',
								name: 'Type: Discussion',
								color: 'fef2c0',
								__typename: 'Label'
							}
						],
						__typename: 'LabelConnection'
					},
					__typename: 'Issue'
				},
				{
					number: 21664,
					title: 'Feedback on startTransition/useTransition syntax',
					state: 'OPEN',
					closedAt: null,
					createdAt: '2021-06-11T15:42:43Z',
					comments: {
						totalCount: 1,
						__typename: 'IssueCommentConnection'
					},
					author: {
						login: 'stefanhamburger',
						__typename: 'User'
					},
					labels: {
						totalCount: 2,
						nodes: [
							{
								id: 'MDU6TGFiZWwzMDc5MDUzOTE5',
								name: 'React 18',
								color: '26B0E4',
								__typename: 'Label'
							},
							{
								id: 'MDU6TGFiZWw3MTAzNzU3OTI=',
								name: 'Type: Discussion',
								color: 'fef2c0',
								__typename: 'Label'
							}
						],
						__typename: 'LabelConnection'
					},
					__typename: 'Issue'
				},
				{
					number: 21662,
					title: 'Concurrency / time-slicing by default',
					state: 'OPEN',
					closedAt: null,
					createdAt: '2021-06-11T09:32:25Z',
					comments: {
						totalCount: 7,
						__typename: 'IssueCommentConnection'
					},
					author: {
						login: 'rybon',
						__typename: 'User'
					},
					labels: {
						totalCount: 2,
						nodes: [
							{
								id: 'MDU6TGFiZWwzMDc5MDUzOTE5',
								name: 'React 18',
								color: '26B0E4',
								__typename: 'Label'
							},
							{
								id: 'MDU6TGFiZWw3MTAzNzU3OTI=',
								name: 'Type: Discussion',
								color: 'fef2c0',
								__typename: 'Label'
							}
						],
						__typename: 'LabelConnection'
					},
					__typename: 'Issue'
				},
				{
					number: 21645,
					title: 'Backport releases on GitHub have incorrect tarballs',
					state: 'OPEN',
					closedAt: null,
					createdAt: '2021-06-08T09:57:08Z',
					comments: {
						totalCount: 0,
						__typename: 'IssueCommentConnection'
					},
					author: {
						login: 'evoL',
						__typename: 'User'
					},
					labels: {
						totalCount: 1,
						nodes: [
							{
								id: 'MDU6TGFiZWwxNTU5ODQxNjA=',
								name: 'Status: Unconfirmed',
								color: 'd4c5f9',
								__typename: 'Label'
							}
						],
						__typename: 'LabelConnection'
					},
					__typename: 'Issue'
				},
				{
					number: 21636,
					title:
						'[DevTools Bug] Cannot add node "476" because a node with that id is already in the Store.',
					state: 'OPEN',
					closedAt: null,
					createdAt: '2021-06-07T06:52:01Z',
					comments: {
						totalCount: 8,
						__typename: 'IssueCommentConnection'
					},
					author: {
						login: 'Bodyhealer',
						__typename: 'User'
					},
					labels: {
						totalCount: 4,
						nodes: [
							{
								id: 'MDU6TGFiZWw3MTA1NzM1OTU=',
								name: 'Component: Developer Tools',
								color: 'fbca04',
								__typename: 'Label'
							},
							{
								id: 'MDU6TGFiZWw2MjAzNjg0MDc=',
								name: 'Resolution: Needs More Information',
								color: 'fffde7',
								__typename: 'Label'
							},
							{
								id: 'MDU6TGFiZWwxNTU5ODQxNjA=',
								name: 'Status: Unconfirmed',
								color: 'd4c5f9',
								__typename: 'Label'
							},
							{
								id: 'MDU6TGFiZWw0MDkyOTE1MQ==',
								name: 'Type: Bug',
								color: 'b60205',
								__typename: 'Label'
							}
						],
						__typename: 'LabelConnection'
					},
					__typename: 'Issue'
				}
			],
			__typename: 'SearchResultItemConnection'
		}
	}
};
