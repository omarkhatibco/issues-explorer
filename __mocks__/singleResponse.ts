import { RepositoryQueryResponse } from 'types';

interface Response {
	data: RepositoryQueryResponse;
}

export const singleResponse: Response = {
	data: {
		repository: {
			__typename: 'Repository',
			issue: {
				__typename: 'Issue',
				number: 21636,
				title:
					'[DevTools Bug] Cannot add node "476" because a node with that id is already in the Store.',
				createdAt: '2021-06-07T06:52:01Z',
				state: 'OPEN',
				bodyHTML:
					'<h3>Website or app</h3>\n<p><a rel="nofollow" href="https://zhivika.ru/product/26406103_plastyr_perc_bperf_10smh18sm_n">https://zhivika.ru/product/26406103_plastyr_perc_bperf_10smh18sm_n</a></p>\n<h3>Repro steps</h3>\n<ol>\n<li>Load page.</li>\n<li>Open react components.</li>\n<li>Click button "Смотреть все аптеки" for unfold all list.</li>\n</ol>\n<h3>How often does this bug happen?</h3>\n<p>Every time</p>\n<h3>DevTools package (automated)</h3>\n<p>react-devtools-extensions</p>\n<h3>DevTools version (automated)</h3>\n<p>4.13.5-0ae5290b54</p>\n<h3>Error message (automated)</h3>\n<p>Cannot add node "476" because a node with that id is already in the Store.</p>\n<h3>Error call stack (automated)</h3>\n<p><em>No response</em></p>\n<h3>Error component stack (automated)</h3>\n<p><em>No response</em></p>\n<h3>GitHub query string (automated)</h3>\n<div class="snippet-clipboard-content position-relative" data-snippet-clipboard-copy-content="https://api.github.com/search/issues?q=Cannot add node  because a node with that id is already in the Store. in:title is:issue is:open is:public label:&quot;Component: Developer Tools&quot; repo:facebook/react\n"><pre lang="text"><code>https://api.github.com/search/issues?q=Cannot add node  because a node with that id is already in the Store. in:title is:issue is:open is:public label:"Component: Developer Tools" repo:facebook/react\n</code></pre></div>',
				author: {
					__typename: 'User',
					avatarUrl:
						'https://avatars.githubusercontent.com/u/7447214?u=ab95174ef2cd1fb8163a463bd3f9d8b86cb5ad60&v=4',
					login: 'Bodyhealer'
				},
				comments: {
					__typename: 'IssueCommentConnection',
					totalCount: 8,
					nodes: [
						{
							__typename: 'IssueComment',
							id: 'MDEyOklzc3VlQ29tbWVudDg1NTkwMTc1Nw==',
							bodyHTML:
								'<p>Hi <a class="user-mention" data-hovercard-type="user" data-hovercard-url="/users/Bodyhealer/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/Bodyhealer">@Bodyhealer</a>. I\'m no table to reproduce the error following the steps you\'ve described.</p>\n<details open="" class="details-reset border rounded-2">\n  <summary class="px-3 py-2 border-bottom">\n    <svg aria-hidden="true" viewBox="0 0 16 16" version="1.1" data-view-component="true" height="16" width="16" class="octicon octicon-device-camera-video">\n    <path fill-rule="evenodd" d="M16 3.75a.75.75 0 00-1.136-.643L11 5.425V4.75A1.75 1.75 0 009.25 3h-7.5A1.75 1.75 0 000 4.75v6.5C0 12.216.784 13 1.75 13h7.5A1.75 1.75 0 0011 11.25v-.675l3.864 2.318A.75.75 0 0016 12.25v-8.5zm-5 5.075l3.5 2.1v-5.85l-3.5 2.1v1.65zM9.5 6.75v-2a.25.25 0 00-.25-.25h-7.5a.25.25 0 00-.25.25v6.5c0 .138.112.25.25.25h7.5a.25.25 0 00.25-.25v-4.5z"></path>\n</svg>\n    <span aria-label="Video description bug-21636-repro-Kapture.2021-06-07.at.08.56.19.mp4" class="m-1">bug-21636-repro-Kapture.2021-06-07.at.08.56.19.mp4</span>\n    <span class="dropdown-caret"></span>\n  </summary>\n\n  <video src="https://user-images.githubusercontent.com/29597/121020334-47ced780-c76e-11eb-8004-093dddedea3b.mp4" data-canonical-src="https://user-images.githubusercontent.com/29597/121020334-47ced780-c76e-11eb-8004-093dddedea3b.mp4" controls="controls" muted="muted" class="d-block rounded-bottom-2 width-fit" style="max-height:640px;">\n\n  </video>\n</details>\n\n<p>Any suggestions?</p>',
							createdAt: '2021-06-07T12:57:06Z',
							author: {
								__typename: 'User',
								avatarUrl:
									'https://avatars.githubusercontent.com/u/29597?u=417a1ec63c2396239beb9453aafdd54275c94765&v=4',
								login: 'bvaughn'
							}
						},
						{
							__typename: 'IssueComment',
							id: 'MDEyOklzc3VlQ29tbWVudDg2MDE0ODMxOQ==',
							bodyHTML:
								'<p>I\'ve got this error today as well on Firefox 89.<br>\nAdding to OP\'s report, this happens when refreshing the page while the dev console is open with the <code>Components</code> tab shown (see screenshot). Not sure if it happens in other scenarios too.</p>\n<p><a target="_blank" rel="noopener noreferrer" href="https://user-images.githubusercontent.com/15946186/121795165-8cca9200-cbe4-11eb-9f71-972ae97ee0b7.png"><img src="https://user-images.githubusercontent.com/15946186/121795165-8cca9200-cbe4-11eb-9f71-972ae97ee0b7.png" alt="image" style="max-width:100%;"></a></p>\n<p>P.S.: Clicking on <code>Dismiss</code> gets rid of the error and seems to show the components correctly again, but it shows up again if the page is refreshed.</p>',
							createdAt: '2021-06-13T04:13:47Z',
							author: {
								__typename: 'User',
								avatarUrl: 'https://avatars.githubusercontent.com/u/15946186?v=4',
								login: 'rdcavanha'
							}
						},
						{
							__typename: 'IssueComment',
							id: 'MDEyOklzc3VlQ29tbWVudDg2MDgzMDUxNQ==',
							bodyHTML:
								'<p><a class="user-mention" data-hovercard-type="user" data-hovercard-url="/users/rdcavanha/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/rdcavanha">@rdcavanha</a> You\'re seeing this bug using OP\'s codebase? Or a different app?</p>',
							createdAt: '2021-06-14T16:44:12Z',
							author: {
								__typename: 'User',
								avatarUrl:
									'https://avatars.githubusercontent.com/u/29597?u=417a1ec63c2396239beb9453aafdd54275c94765&v=4',
								login: 'bvaughn'
							}
						},
						{
							__typename: 'IssueComment',
							id: 'MDEyOklzc3VlQ29tbWVudDg2MTA1NTQ2Mg==',
							bodyHTML:
								'<blockquote>\n<p><a class="user-mention" data-hovercard-type="user" data-hovercard-url="/users/rdcavanha/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/rdcavanha">@rdcavanha</a> You\'re seeing this bug using OP\'s codebase? Or a different app?</p>\n</blockquote>\n<p>Completely different app.</p>',
							createdAt: '2021-06-14T23:20:38Z',
							author: {
								__typename: 'User',
								avatarUrl: 'https://avatars.githubusercontent.com/u/15946186?v=4',
								login: 'rdcavanha'
							}
						},
						{
							__typename: 'IssueComment',
							id: 'MDEyOklzc3VlQ29tbWVudDg2MTU1MTg3Nw==',
							bodyHTML:
								'<p><a class="user-mention" data-hovercard-type="user" data-hovercard-url="/users/rdcavanha/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/rdcavanha">@rdcavanha</a> Could you share your code then? <em>I\'m</em> still looking for a way to repro this issue.</p>',
							createdAt: '2021-06-15T14:33:15Z',
							author: {
								__typename: 'User',
								avatarUrl:
									'https://avatars.githubusercontent.com/u/29597?u=417a1ec63c2396239beb9453aafdd54275c94765&v=4',
								login: 'bvaughn'
							}
						},
						{
							__typename: 'IssueComment',
							id: 'MDEyOklzc3VlQ29tbWVudDg2MTkxMzU2NA==',
							bodyHTML:
								'<p><a class="user-mention" data-hovercard-type="user" data-hovercard-url="/users/bvaughn/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/bvaughn">@bvaughn</a> Sorry, I can\'t share the code as it is a private repo.</p>\n<p>However, I\'ve noticed that after I refreshed my Firefox profile the issue seems to be gone. I\'ve tried using the React extension on the same app as before and was unable to reproduce the bug again.</p>\n<p>This leads me to believe that it\'s probably some sort of conflict with another extension (just a guess). Unfortunately I can\'t say which one exactly as I no longer have all the extensions installed due to the profile refresh. The fact that this has very low occurrence seems to support this idea of extension conflict as well.</p>',
							createdAt: '2021-06-16T00:08:50Z',
							author: {
								__typename: 'User',
								avatarUrl: 'https://avatars.githubusercontent.com/u/15946186?v=4',
								login: 'rdcavanha'
							}
						},
						{
							__typename: 'IssueComment',
							id: 'MDEyOklzc3VlQ29tbWVudDg2MjUzMzU2Mg==',
							bodyHTML: '<p>Thanks for the extra info.</p>',
							createdAt: '2021-06-16T16:39:31Z',
							author: {
								__typename: 'User',
								avatarUrl:
									'https://avatars.githubusercontent.com/u/29597?u=417a1ec63c2396239beb9453aafdd54275c94765&v=4',
								login: 'bvaughn'
							}
						},
						{
							__typename: 'IssueComment',
							id: 'MDEyOklzc3VlQ29tbWVudDg2MjU3MTUzOQ==',
							bodyHTML:
								'<p>[Win10 64-bit] [Firefox 89.0]<br>\nGot the issue today, you may be able to replicate it with the following steps :</p>\n<ol>\n<li>If the React devtools extension is installed, uninstall it</li>\n<li>Go on a web page that uses React and open the Firefox devtools</li>\n<li>In a new/different tab, install the React devtools extension</li>\n<li>Keep the Firefox devtools opened and refresh your page (CTRL + F5)</li>\n<li>Go to the Components tab</li>\n</ol>',
							createdAt: '2021-06-16T17:30:01Z',
							author: {
								__typename: 'User',
								avatarUrl:
									'https://avatars.githubusercontent.com/u/41510224?u=1f9529391466592d46e9c71b7a2141dd9cded5b9&v=4',
								login: 'Romej'
							}
						}
					]
				}
			}
		}
	}
};
