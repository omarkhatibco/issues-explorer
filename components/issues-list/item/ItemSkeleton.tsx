import styled from 'styled-components';

export const ItemSkeleton: React.FC = () => {
	return (
		<Wrapper role='group'>
			<Article>
				<IconWrapper>
					<Icon />
				</IconWrapper>
				<ContentWrapper>
					<Heading />
					<Paragraph />
				</ContentWrapper>
			</Article>
		</Wrapper>
	);
};

const Wrapper = styled.li`
	background-color: var(--color-white);
	cursor: pointer;
	transition: var(--transition);
	padding-block: 1rem;
	padding-inline: 1rem;

	&:hover {
		background-color: var(--color-gray);
	}
`;

const Article = styled.article`
	width: 100%;
	display: flex;

	& > *:not(:last-child) {
		padding-inline-end: 0.5rem;
	}
`;

const IconWrapper = styled.div``;
const ContentWrapper = styled.div`
	flex-grow: 1;
`;

const Icon = styled.div`
	width: 1.25rem;
	height: 1.25rem;
	border-radius: 100%;
	background-color: var(--color-border);
	animation: 0.8s linear 0s infinite alternate none running animation-skeleton;
`;

const Heading = styled(Icon)`
	width: 66%;
	height: 1rem;
	border-radius: 0.25rem;
	margin-inline-end: 0.25rem;
	padding-block: 0.25rem;
`;
const Paragraph = styled(Heading)`
	width: 33%;
	height: 0.75rem;
	margin-block-start: 0.25rem;
`;
