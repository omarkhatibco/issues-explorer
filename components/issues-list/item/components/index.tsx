export * from './ItemCommentsCount';
export * from './ItemContent';
export * from './ItemContext';
export * from './ItemStatus';
