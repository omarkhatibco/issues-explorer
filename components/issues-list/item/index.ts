export * from './Item';
export * from './ItemSkeleton';
