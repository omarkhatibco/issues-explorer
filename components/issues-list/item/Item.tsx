import styled from 'styled-components';
import { Issue } from 'types';
import { ItemCommentsCount, ItemContent, ItemContext, ItemStatus } from './components';

export const Item: React.FC<Issue> = props => {
	return (
		<ItemContext.Provider value={props}>
			<Wrapper role='group'>
				<Article>
					<ItemStatus />
					<ItemContent />
					<ItemCommentsCount />
				</Article>
			</Wrapper>
		</ItemContext.Provider>
	);
};

const Wrapper = styled.li`
	background-color: var(--color-white);
	cursor: pointer;
	transition: var(--transition);
	padding-block: 1rem;
	padding-inline: 1rem;

	&:hover {
		background-color: var(--color-gray);
	}
`;

const Article = styled.article`
	width: 100%;
	display: flex;

	& > *:not(:last-child) {
		padding-inline-end: 0.5rem;
	}
`;
