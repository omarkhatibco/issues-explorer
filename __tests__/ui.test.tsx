import { act, render, screen } from '@testing-library/react';
import { Item } from 'components/issues-list/item/Item';
import React from 'react';
import { response } from '__mocks__';

describe('Single Issue', () => {
	const props = response.data.search.nodes[0];

	it('matches snapshot', async () => {
		await act(async () => {
			const { container } = render(<Item {...props} />);

			expect(container).toMatchSnapshot();
		});
	});

	it('renders elements correctly', async () => {
		render(<Item {...props} />);
		const title = screen.getByTestId('title');
		expect(title).toHaveTextContent(props.title);

		const commentsCount = screen.getByTestId('comments-count');
		expect(commentsCount).toHaveTextContent(`${props.comments.totalCount}`);
	});
});
