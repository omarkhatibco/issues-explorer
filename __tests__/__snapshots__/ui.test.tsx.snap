// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Issue matches snapshot 1`] = `
<div>
  <li
    class="Item__Wrapper-sc-1fnnn89-0 iBKZwl"
    role="group"
  >
    <article
      class="Item__Article-sc-1fnnn89-1 leRLLF"
    >
      <div
        class="ItemStatus__Wrapper-sc-1x9kk55-0 dVyFut"
        title="Open issue"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 14 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="ItemContent__Wrapper-sc-1255c34-0 ecZkjT"
      >
        <div
          class="ItemContent__TitleWrapper-sc-1255c34-1 bRqJbF"
        >
          <h1
            class="ItemContent__Title-sc-1255c34-4 fJkKbV"
            data-testid="title"
          >
            [DevTools Bug]: Long keys overlap
          </h1>
          <div
            class="ItemContent__LabelsWrapper-sc-1255c34-2 keeNLA"
          >
            <label
              class="ItemContent__Label-sc-1255c34-3 jYOrxc"
            >
              Component: Developer Tools
            </label>
            <label
              class="ItemContent__Label-sc-1255c34-3 dnboLi"
            >
              Resolution: Needs More Information
            </label>
            <label
              class="ItemContent__Label-sc-1255c34-3 kdbzH"
            >
              Status: Unconfirmed
            </label>
            <label
              class="ItemContent__Label-sc-1255c34-3 hAZNUh"
            >
              Type: Bug
            </label>
          </div>
        </div>
        <p
          class="ItemContent__Info-sc-1255c34-5 dXCxEV"
        >
          #
          21695
           
          Opened
           
          <time
            datetime="2021-06-16T20:57:00Z"
            title="6/16/2021, 10:57:00 PM"
          >
            in 2 days
          </time>
           
          by 
          MinimumViablePerson
        </p>
      </div>
      <div
        class="ItemCommentsCount__Wrapper-bdjqbc-0 juLnoJ"
        title="1 comment"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="ItemCommentsCount__Count-bdjqbc-1 kIbysp"
          data-testid="comments-count"
        >
          1
        </span>
      </div>
    </article>
  </li>
</div>
`;
